src/ScottPlot4/ScottPlot/Renderable/Legend.cs

```diff
@@ -191,6 +220,88 @@ public Bitmap GetBitmap(bool lowQuality = false, double scale = 1.0)
             }
         }
 
+        private void RenderOnBitmapHorizontal(Graphics gfx, LegendItem[] items, System.Drawing.Font font,
```

Comment:
This function and `RenderOnBitmapVertical` have a lot of commonality, I think they can be refactored so that we're not repeating ourselves quite so much. All of the code for actual drawing is the complete same, the only difference is where we do that drawing.

We could refactor this to have functions like `DrawLegendItem`, which could be used by both horizontal and vertical legends, they would just call them with different coordinates.

For example, in ScottPlot 5, we have [`RenderLegendItem`](https://github.com/ScottPlot/ScottPlot/blob/main/src/ScottPlot5/ScottPlot5/Plottables/Legend.cs#L64-L90), which could be reused for horizontal legends (though they aren't supported in v5).

Reply:
FWIW, we probably should've refactored it to separate rendering the individual legend items from the legend as a whole, so this is more of a preexisting issue that became more important because of your PR, rather than anything that your PR introduced.

Reply:
Refactored on commit https://github.com/ScottPlot/ScottPlot/pull/2216/commits/437a273ddd6184cdd33138009e33e2e608c53a10.

Now RenderOnBitmap sets the offsets properly and calls the common method DrawLegendItem

(LucaBattEL it's always me, I committed with the wrong account, sorry)